Sources/Core/Renderer/GUI/GUIElement/GUITextInput.swift

```diff
@@ -12,21 +13,43 @@ struct GUITextInput: GUIElement {
     ).meshes(context: context)
 
     // Message
-    var message = try content.meshes(context: context)
-    message.translate(amount: [2, 2])
+    var messageBeforeCursor = String(content.prefix(upTo: cursorIndex))
```

Comment:
messageBeforeCursor and messageAfterCursor can probably be `let` (unless there's some mutation somewhere that I'm missing).